modules/study/src/main/LightStudyCache.scala

```diff
@@ -0,0 +1,34 @@
+package lila.study
+
+import com.github.blemale.scaffeine.{ AsyncLoadingCache, Scaffeine }
+import scala.concurrent.duration._
+
+import lila.study._
+import lila.user.User
+
+final class LightStudyCache(ttl: Duration,
+	                        studyRepo: StudyRepo,
+	                        asyncCache: lila.memo.AsyncCache.Builder) {
+	private val cache = asyncCache.clearable(
+	  name = "study.lightStudyCache",
+    f = fetch,
+    expireAfter = _.ExpireAfterWrite(4 hours)
+	)
+
+    def remove(studyId: String): Unit =
+      cache invalidate studyId
+
+    def get(studyId: String): Fu[Option[LightStudy]] =
+      cache get studyId
+
+    private def fetch(studyId: String): Fu[Option[LightStudy]] =
```

Comment:
use `Study.Id` instead of `String`. Always use specific types when possible, for greater safety.